tensorboard/components/tf_ng_tensorboard/app.component.ts

```diff
@@ -13,10 +13,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 import {Component} from '@angular/core';
-import {Store, select} from '@ngrx/store';
-import {State, getActivePlugin} from './core/core.reducers';
-import {changePlugin} from './core/core.actions';
-import {PluginId} from './types/api';
+import {Store} from '@ngrx/store';
+import {State} from './core/core.reducers';
+import {coreLoaded} from './core/core.actions';
```

Comment:
Nope, respective plugin will have their own action for mutating the store. Partially because of lazy loading, it looks like the it is convention to create a module of action/reducer/component out of a feature which, for us, would be plugins. As such, I foresee us having a separate actions per plugin. 

This is not to say other features won't have dependency on core; when a plugin want to switch dashboard programmatically, it would call the core action.